recipe/unzip_test.go

```diff
@@ -0,0 +1,171 @@
+package main_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/julz/cube/recipe"
+)
+
+var _ = Describe("Unzip function", func() {
+
+	var (
+		targetDir string
+		srcZip    string
+		err       error
+	)
+
+	JustBeforeEach(func() {
+		err = Unzip(srcZip, targetDir)
+	})
+
+	Context("Unzip secceeds", func() {
+
+		zippedFiles := map[string]string{
+			"file1":                       "this is the content of test file 1",
+			"innerDir/file2":              "this is the content of test file 2",
+			"innerDir/innermostDir/file3": "this is the content of test file 3",
+		}
+
+		getRoot := func(path string) string {
+			pathParts := strings.Split(path, "/")
+			return pathParts[0]
+		}
+
+		removeFile := func(file string) {
+			if err := os.RemoveAll(file); err != nil {
+				panic(err)
+			}
+		}
+
+		cleanUpFiles := func() {
+			for filePath := range zippedFiles {
+				rootDir := getRoot(filePath)
+				removeFile(rootDir)
+			}
+		}
+
+		assertFileContents := func(file string, expectedContent string) {
+			path := filepath.Join(targetDir, file)
+			content, err := ioutil.ReadFile(path)
+			if err != nil {
+				panic(err)
+			}
+			Expect(content).To(Equal([]byte(expectedContent)))
+		}
+
+		assertFilesUnzippedSuccessfully := func() {
+			It("should not fail", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should unzip the files in the target directory", func() {
+				for fileName := range zippedFiles {
+					path := filepath.Join(targetDir, fileName)
+					Expect(path).To(BeAnExistingFile())
+				}
+			})
+
+			It("should not change file contents", func() {
+				for file, expectedContent := range zippedFiles {
+					assertFileContents(file, expectedContent)
+				}
+			})
+		}
+
+		Context("When target directory is not specified", func() {
```

Comment:
I think the target directory should be mandatory. For this test that means when a target directory is not specified it should return an error rather than assuming a directory to unzip to.

Reply:
Ok, making it mandatory would make it read a bit better. Which path should we use in the invocation of Unzip in recipe.go &mdash; `"."` or `"/workspace"`?